jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/prefixes/PrefixesRDF.java

```diff
@@ -0,0 +1,220 @@
+                            prefixes:prefixName  "prefix1" ;
+                            prefixes:prefixURI   "http://www.localhost.org/uri1#"^^xsd:anyURI
+                   }""";
+            UpdateExec.dataset(dataset).update(update1).execute();
+
+            String update2 = """
+                    PREFIX prefixes: <http://jena.apache.org/prefixes#>
+                    PREFIX rdf:      <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+                    PREFIX xsd:      <http://www.w3.org/2001/XMLSchema#>
+                    
+                    INSERT DATA {
+                          [] rdf:type prefixes:Prefix ;
+                             prefixes:prefixName  "prefix2" ;
+                             prefixes:prefixURI   "http://www.localhost.org/uri2#"^^xsd:anyURI
+                    }""";
+            UpdateExec.dataset(dataset).update(update2).execute();
+
+            dataset.commit();
+        } catch (Throwable th) {
+            dataset.abort();
+            throw(th);
+        } finally {
+            dataset.end() ;
+        }
+
+    }
+
+    public Transactional transactional() { return dataset; }
+
+    @Override
+    public Optional<String> fetchURI(String prefix) {
+
+        String query = "PREFIX prefixes: <http://jena.apache.org/prefixes#> " +
+                "SELECT ?prefixURI WHERE { " +
+                "?X prefixes:prefixName \"" + prefix + "\" . " +
+                "?X prefixes:prefixURI ?prefixURI " +
+                "}";
+
+        try (QueryExec qExec = QueryExec.dataset(dataset).query(query).build()) {
+
+            RowSet rowSet = qExec.select();
+            List<String> x = new ArrayList<String>();
+
+            rowSet.forEachRemaining(row -> {
+                String prefixNameLiteral = row.get("prefixURI").getLiteralLexicalForm();
+                if (prefixNameLiteral != null) {
+                    x.add(prefixNameLiteral);
+                }
+            });
+            if (x.isEmpty())
+                return Optional.empty();
+            //originally x.getFirst()
+            return Optional.ofNullable(x.get(0));
+        }
+    }
+
+    @Override
+    public void updatePrefix(String prefix, String uri) {
+
+        String query =
+                "PREFIX prefixes: <http://jena.apache.org/prefixes#>\n" +
+                "PREFIX xsd:      <http://www.w3.org/2001/XMLSchema#>\n" +
+                "ASK WHERE {  ?X prefixes:prefixName  \"" + prefix + "\" ; \n" +
+                "                prefixes:prefixURI   ?prefixURI . }";
+        try (QueryExec qExec = QueryExec.dataset(dataset).query(query).build()) {
+
+            AtomicBoolean result = new AtomicBoolean(false);
+            dataset.executeRead(()->{
+                result.set(qExec.ask());
+            });
+
+            String update;
+            if (result.get())
+                update =
+                        "PREFIX prefixes: <http://jena.apache.org/prefixes#>\n" +
```

Comment:
Stylistic: Maybe use multi-line strings to make the queries and updates more readable in the code?